tests/unit_tests/pyanaconda_tests/modules/boss/test_copy_logs_task.py

```diff
@@ -0,0 +1,124 @@
+#
+# Copyright (C) 2021  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os.path
+import tempfile
+import unittest
+from unittest.mock import call, patch
+from pyanaconda.modules.boss.installation import CopyLogsTask
+
+
+class CopyLogsTaskTest(unittest.TestCase):
+    @patch("pyanaconda.modules.boss.installation.glob.glob")
+    @patch("pyanaconda.modules.boss.installation.util.execInSysroot")
+    @patch("pyanaconda.modules.boss.installation.util.execWithRedirect")
+    @patch("pyanaconda.modules.boss.installation.flags")
+    def test_run_all(self, flags_mock, exec_wr_mock, exec_isr_mock, glob_mock):
+        """Test the log copying task."""
+        glob_mock.return_value = ["/tmp/ks-script-blabbityblah.log"]
+        flags_mock.nosave_logs = False
+        flags_mock.nosave_input_ks = False
+
+        with tempfile.TemporaryDirectory() as dest:
```

Comment:
You can also use `tmp_path` fixture nicely here if you want:
https://docs.pytest.org/en/latest/reference/reference.html#std-fixture-tmp_path

Reply:
Thank you for the review! I'm still trying to make the tests work, so likely I'll change it more. I want to patch also the dir creation and do nothing to the filesystem.

Reply:
Maybe the best option would be to patch functions in the shutil and not your functions. You will test more code by that.

Reply:
Well, yes and no. The way tests are written now, filesystem is not touched. If I patched directly `copyfile` and `copytree`, I'd need also `os.path.exists`, and then we're at 8 patches per test. That sounds wrong.